### ISSUE TITLE
init.fish causes $PATH variable to grow every time prompt is printed

### ISSUE THREAD

**Issue by silenc3r** [ocaml/opam#4524]
Sourcing `init.fish` in fish config file causes $PATH variable to be prepended with `.` (dot) every time prompt is displayed.
```
$ echo $PATH
. /home/dawid/src/projects/zummi/_opam/bin /home/dawid/bin /usr/lib64/ccache /usr/local/bin /usr/local/sbin /usr/bin /usr/sbin
$ echo $PATH
. /home/dawid/src/projects/zummi/_opam/bin . /home/dawid/bin /usr/lib64/ccache /usr/local/bin /usr/local/sbin /usr/bin /usr/sbin
$ echo $PATH
. /home/dawid/src/projects/zummi/_opam/bin . . /home/dawid/bin /usr/lib64/ccache /usr/local/bin /usr/local/sbin /usr/bin /usr/sbin
$ echo $PATH
. /home/dawid/src/projects/zummi/_opam/bin . . . /home/dawid/bin /usr/lib64/ccache /usr/local/bin /usr/local/sbin /usr/bin /usr/sbin
```
Without sourcing init.fish:
```
$ echo $PATH
/home/dawid/bin /usr/lib64/ccache /usr/local/bin /usr/local/sbin /usr/bin /usr/sbin
$ echo $PATH
/home/dawid/bin /usr/lib64/ccache /usr/local/bin /usr/local/sbin /usr/bin /usr/sbin
$ echo $PATH
/home/dawid/bin /usr/lib64/ccache /usr/local/bin /usr/local/sbin /usr/bin /usr/sbin

```

Also `opam env --shell=fish` prepends path to $OPAMROOT/default/bin with `:` (colon), eg.:
```
$ opam env --shell=fish
set -gx OPAM_SWITCH_PREFIX '/home/dawid/.local/opam/default';
set -gx CAML_LD_LIBRARY_PATH '/home/dawid/.local/opam/default/lib/stublibs:/home/dawid/.local/opam/default/lib/ocaml/stublibs:/home/dawid/.local/opam/default/lib/ocaml';
set -gx OCAML_TOPLEVEL_PATH '/home/dawid/.local/opam/default/lib/toplevel';
builtin -n | /bin/sh -c 'grep -q \'^argparse$\'' 1>/dev/null 2>/dev/null; and set -gx MANPATH ':/home/dawid/.local/opam/default/man';
set -gx PATH ':/home/dawid/.local/opam/default/bin' '/home/dawid/bin' '/usr/lib64/ccache' '/usr/local/bin' '/usr/local/sbin' '/usr/bin' '/usr/sbin';
$ opam env --shell=fish
set -gx OPAM_SWITCH_PREFIX '/home/dawid/.local/opam/default';
set -gx CAML_LD_LIBRARY_PATH '/home/dawid/.local/opam/default/lib/stublibs:/home/dawid/.local/opam/default/lib/ocaml/stublibs:/home/dawid/.local/opam/default/lib/ocaml';
set -gx OCAML_TOPLEVEL_PATH '/home/dawid/.local/opam/default/lib/toplevel';
builtin -n | /bin/sh -c 'grep -q \'^argparse$\'' 1>/dev/null 2>/dev/null; and set -gx MANPATH ':/home/dawid/.local/opam/default/man';
set -gx PATH ':/home/dawid/.local/opam/default/bin' '.' '/home/dawid/bin' '/usr/lib64/ccache' '/usr/local/bin' '/usr/local/sbin' '/usr/bin' '/usr/sbin';
$ opam env --shell=fish
set -gx OPAM_SWITCH_PREFIX '/home/dawid/.local/opam/default';
set -gx CAML_LD_LIBRARY_PATH '/home/dawid/.local/opam/default/lib/stublibs:/home/dawid/.local/opam/default/lib/ocaml/stublibs:/home/dawid/.local/opam/default/lib/ocaml';
set -gx OCAML_TOPLEVEL_PATH '/home/dawid/.local/opam/default/lib/toplevel';
builtin -n | /bin/sh -c 'grep -q \'^argparse$\'' 1>/dev/null 2>/dev/null; and set -gx MANPATH ':/home/dawid/.local/opam/default/man';
set -gx PATH ':/home/dawid/.local/opam/default/bin' '.' '.' '/home/dawid/bin' '/usr/lib64/ccache' '/usr/local/bin' '/usr/local/sbin' '/usr/bin' '/usr/sbin';
```
This only happens for default switch.
Edit: after using workaround for dots in path this applies to all switches.

Opam config report:
```
# opam config report
# opam-version      2.0.6
# self-upgrade      no
# system            arch=x86_64 os=linux os-distribution=fedora os-version=33
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              3
# repositories      1 (http) (default repo at 71479db1)
# pinned            0
# current-switch    default
[WARNING] PATH contains '.' : this is a likely cause of trouble.
```

**Comment by silenc3r** [comment]
`$OPAMROOT` is set to `/home/dawid/.local/opam`

**Comment by silenc3r** [comment]
Adding this to `config.fish` is a simple workaround
```
function __filter_path
    for p in $PATH
        if [ $p != '.' ]
            echo $p
        end
    end
end

function filter_path --on-event fish_prompt;
    set -gx PATH (__filter_path)
end
```

**Comment by rjbou** [comment]
The colon issue is resolved by #4084, that is available in 2.1.0~alpha, and be included in 2.0.8

**Comment by silenc3r** [comment]
I've rebuilt opam-2.0.6 with fix from https://github.com/ocaml/opam/pull/4084 and I can confirm it solves the issue with $PATH variable.
